### PR TITLE
feat(oci): generate SPDX SBOMs during import

### LINKED ISSUE #200: feat(oci): add license/SBOM detection for OCI imports
## Summary

- First priority; #199 follows with compliance-bundle downloads
- The compliance pipeline currently only covers the 3 built-in M2Images (Alpine/Ubuntu/Rocky)
- When a user pulls and boots an arbitrary OCI image, there is no way to check the licenses of the packages inside it
- Reference: #190 (the PR that implemented the M2Image compliance pipeline; OCI was explicitly out of scope there), #199 (follow-up download work)

## Architecture

```mermaid
flowchart TD
    Ref[OCI reference] --> Import[OCI import pipeline]
    Import --> Rootfs[Merged rootfs]
    Rootfs --> Detect[Detect package manager apk-dpkg-rpm]
    Detect --> SBOM[Generate SPDX SBOM]
    SBOM --> Template[Attach to registered template]
```

## Scope (MVP)

- Complete before starting #199
- Auto-detect the package manager (apk/dpkg/rpm) in the rootfs at import time
- On successful detection, generate an SPDX SBOM and attach it to the registered template
- On detection failure (unknown distribution), allow the import without an SBOM, with a warning only

### PR DESCRIPTION
**Pull request by mcc0nnell**
Closes #200.

This extends the compliance pipeline to arbitrary OCI imports at the merged-rootfs boundary, before FireCrab mutates the guest filesystem.

What changes:
- auto-detect apk, dpkg, or modern RPM SQLite package databases in the merged OCI rootfs;
- parse package state as data instead of executing binaries from the imported image;
- generate an SPDX 2.3 document with installed package name/version/architecture plus package-manager license/source metadata when available;
- persist successful evidence at `images/compliance/<alias>-<arch>/sbom.spdx.json`, giving #199 a stable bundle location to expose through the API/UI;
- treat unknown package managers or SBOM-generation failures as warning-only so scratch/distroless/custom images still import;
- remove a newly written compliance bundle if template registration subsequently fails;
- reject symlinked package-database paths rather than following imported links outside the merged rootfs;
- keep package-database scanning off the Tokio async worker path;
- accept dpkg records whose state field is `installed`, including held/reinstall-required packages.

Validation:
- compliance unit tests cover apk, dpkg, RPM SQLite/header parsing, malformed metadata, symlink safety, unknown-package-manager behavior, and bundle lifecycle;
- the edge-case coverage suite runs inside the binary crate so llvm-cov attributes execution to the production compliance module;
- `cargo clippy --workspace --all-targets` and the repository Clippy warning gate pass on the current branch;
- full CI, release-profile tests, and Codecov run on every pushed revision.

The branch now carries the current `main` Clippy warning baseline unchanged, so the effective feature diff is limited to the OCI importer/compliance implementation and OCI regression coverage.

## Summary by CodeRabbit

- **New Features**
  - OCI image imports now generate SPDX 2.3 software bills of materials (SBOMs) from APK, dpkg, or RPM package data.
  - Generated SBOMs are stored with the imported image for later compliance review.
  - Package metadata includes licenses, sources, versions, and stable package identifiers.

- **Bug Fixes**
  - Unsupported or missing package databases no longer prevent image imports.
  - SBOM data is cleaned up when image registration fails.

### CODE REVIEW

File: firecrab-api/src/oci.rs [hunk 1]
```diff
@@ -3327,6 +3329,40 @@ async fn import_oci_image_in_scratch(
...
+        }
+        Err(error) => {
+            tracker.append_log(
+                alias,
+                format!("warning: OCI SBOM generation failed: {error}; importing without an SBOM"),
+            );
+            tracing::warn!(alias, error = %error, "OCI SBOM generation failed; continuing import");
+            None
+        }
+    };
```
[the remark this answers is not in the document]
Comment (mcc0nnell) [reply firecrab-api/src/oci.rs:3364]: Addressed in 09ddde0. I kept the compliance scanner's synchronous API, but wrapped its filesystem/SQLite work in `tokio::task::block_in_place`, so the Tokio multi-thread runtime can replace the worker while the scan runs. This avoids adding ownership/plumbing churn to the importer while fixing the worker-starvation issue.

File: firecrab-api/src/oci/compliance.rs [hunk 2]
```diff
@@ -0,0 +1,694 @@
...
+            let Some((key, value)) = line.split_once(':') else {
+                continue;
+            };
+            let key = key.to_owned();
+            fields.insert(key.clone(), value.trim().to_owned());
+            current = Some(key);
+        }
+        if fields.get("Status").map(String::as_str) != Some("install ok installed") {
+            continue;
+        }
```
[the remark this answers is not in the document]
Comment (mcc0nnell) [reply firecrab-api/src/oci/compliance.rs]: Addressed in 09ddde0. The dpkg parser now validates an exact three-field `Status` and keys only on the third/state field being `installed`, regardless of want/error. The regression fixture covers `install ok installed`, `hold ok installed`, `install reinstreq installed`, removed packages, and malformed status shapes.

**Comment by SteelCrab** [comment]
@mcc0nnell 
Thanks for the great work on this! Looks solid overall. Once the merge conflict in .github/clippy-warning-baseline.json is resolved, I'll go ahead and merge it.
